source/common/http/conn_manager_impl.cc

```diff
@@ -259,7 +259,14 @@ RequestDecoder& ConnectionManagerImpl::newStream(ResponseEncoder& response_encod
   }
 
   ENVOY_CONN_LOG(debug, "new stream", read_callbacks_->connection());
-  ActiveStreamPtr new_stream(new ActiveStream(*this, response_encoder.getStream().bufferLimit()));
+
+  // Set the account to start accounting.
+  Buffer::BufferMemoryAccountSharedPtr downstream_request_account =
+      std::make_shared<Buffer::BufferMemoryAccountImpl>();
```

Comment:
Does creating a new account in each call to newStream mean that each stream ends up using a different account?  Do we want to share accounts across all downstream requests?  I was thinking that we wanted the account to be owned by the ConnectionManagerImpl

Reply:
Inside the guts of this pipeline does get disorientating. 

No the 4 buffers along the H2 path share the same 1 account which is tied to the downstream request. This gets invoked in `ServerConnectionImpl::onBeginHeaders` as a new stream is created from the downstream.

Reply:
If you get 2 requests on the same H2 connection, do they end up with the same or different accounts?

Reply:
Answer is yes, and that's what we're going for.  ok!